src/vdc_collection.rs

```diff
@@ -0,0 +1,130 @@
+use super::storage_manager::*;
+use serde_derive::{Deserialize, Serialize};
+
+/// Supported credential formats.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialFormat {
+    MsoMdoc,
+    JwtVcJson,
+    LdpVc,
+    Other(String), // For ease of expansion.
+}
+
+/// Supported credential types.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialType {
+    Iso18013_5_1mDl,
+    VehicleTitle,
+    Other(String), // For ease of expansion.
+}
+
+/// An individual credential.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Credential {
+    id: String, // Probably a UUID, perhaps we should make it one?
+    format: CredentialFormat,
+    ctype: CredentialType,
+    payload: Vec<u8>, // The actual credential.
+}
+
+impl Credential {
+    /// Create a new credential.
+    fn new(
+        id: &str,
+        format: CredentialFormat,
+        ctype: CredentialType,
+        payload: Vec<u8>,
+    ) -> Credential {
+        Credential {
+            id: id.to_string(),
+            format,
+            ctype,
+            payload,
+        }
+    }
+}
+
+/// Verifiable Digital Credential Collection
+///
+/// This is the main interface to credentials.
+pub struct VdcCollection {
+    storage: Box<dyn StorageManagerInterface>,
+}
+
+impl VdcCollection {
+    /// Create a new credential set.
+    pub fn new(engine: Box<dyn StorageManagerInterface>) -> VdcCollection {
+        VdcCollection { storage: engine }
+    }
+
+    /// Add a credential to the set.
+    pub fn add(
+        &mut self,
```

Comment:
You should be able to remove `mut` from all of these methods.

Reply:
Done.